### PR TITLE
This pr adds java support of redis.

### PR DESCRIPTION
**Pull request by garimaaroraa**

### CODE REVIEW

File: tests/test_uncommenter.rb [hunk 1]
```diff
@@ -79,6 +79,32 @@
  return (a + b)
 """
 
+SAMPLE_JAVA_COMMENTED = """
```
Comment (sherubthakur) [inline tests/test_uncommenter.rb:82]: I would say that one test per comment style should be sufficient.

**Comment by garimaarorashuttl** [comment]
I have added a simple java project. I am not sure what external dependencies are required to complete redis challenge, is there a way to attempt this challenge  in java before releasing it?

File: tests/test_uncommenter.rb (hunk 1 above)
Comment (rohitpaulk) [reply tests/test_uncommenter.rb:82]: I think we should stick with one test per language as the contract is based on language, and the fact that shared code powers this is an internal detail of the class being tested 👍

**Comment by rohitpaulk** [comment]
> I am not sure what external dependencies are required to complete redis challenge, is there a way to attempt this challenge in java before releasing it?

Yep, I'll put this up on staging for you to test soon @garimaarorashuttl 🙂

File: compiled_starters/redis-starter-java/spawn_redis_server.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+set -e
+mkdir -p out
+javac -sourcepath src src/Main.java -d out
+jar cf java-redis.jar -C out/ .
+rm -rf out
+java -cp java-redis.jar Main "$@"
```
Comment (rohitpaulk) [inline compiled_starters/redis-starter-java/spawn_redis_server.sh:7]: To ensure that signals are passed properly to the child process, we'll need to prepend `exec` to the last command here.

Example: https://github.com/codecrafters-io/languages/blob/5746db439eae64c99189ce464509f0d4a747f2b1/compiled_starters/git-starter-ruby/your_git.sh#L2

(Just realizing that this is something we could add to the contributing guide 🙂) 

Also, can we change `out` to a temporary directory created with `mktemp -d`? This way, even if the command fails halfway, a user won't end up with the `out` directory polluting their git status. This is assuming that the directory doesn't contain any human-readable stuff. If it does, then maybe sticking with `out` is fine 👍

File: compiled_starters/redis-starter-java/src/Main.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class Main {
+  public static void main(String[] args){
+    System.out.println("Your code goes here!");
+
+    //  Uncomment this block to pass the first stage
+    //    ServerSocket serverSocket = null;
+    //    Socket socket = null;
```
Comment (rohitpaulk) [inline compiled_starters/redis-starter-java/src/Main.java:11]: Would you be okay with renaming `socket` to `clientSocket`? I think that'll make the intention a bit more obvious 🙂

**Comment by rohitpaulk** [comment]
Turns out Java is in high demand 🙂 The top comment from [this reddit post](https://www.reddit.com/r/learnprogramming/comments/l20t2q/the_build_your_own_redis_challenge/): 

Screenshot 2021-01-22 at 6 29 42 PM

I'll pick this up soon and rebase on master.

**Pull request merged**